admin_guide/scheduling/scheduler.adoc

```diff
@@ -0,0 +1,827 @@
++
+----
+admissionConfig:
+  pluginConfig:
+    PodNodeSelector:
+      configuration:
+        apiversion: v1
+        kind:podNodeSelectorPluginConfig: <1>
+          clusterDefaultNodeSelector: region=west <2>
+          ns1: region=west,env=test,infra=fedora,os=fedora <3>
+----
++
+<1> Adds the PodNodeSelector* admission controller plug-in.
+<2> Creates a list of allowed labels for the `ns1` project. 
+<3> Creates a whitelist of permitted labels in the specified project. Here, the project is `ns1` and the labels are the `key=value` pairs that follow.
++
+[NOTE]
+====
+If a project does not have a node selectors specified, the pods associated with that project will be merged with clusterDefaultNodeSelector.
+====
+
+. Restart {product-title} for the changes to take effect.
+ifdef::openshift-origin[]
++
+----
+# systemctl restart origin-master
+----
+endif::[]
+ifdef::openshift-enterprise[]
++
+----
+# systemctl restart atomic-openshift-master
+----
+endif::[]
+
+
+. Create a project object that includes the 
+`scheduler.alpha.kubernetes.io/node-selector` annotation and labels. 
++
+----
+{
+    "kind": "Namespace",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "ns1",
+        "annotations": {
+            "scheduler.alpha.kubernetes.io/node-selector": "env=test,infra=fedora" <1>
+        }
+    },
+    "spec": {},
+    "status": {}
+}
+
+----
++
+<1> Annotation to create the labels to match the project label selector.
+
+. Create a pod specification that includes the labels in the node selector, for example:
++
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    name: hello-pod
+  name: hello-pod
+spec:
+  containers:
+    - image: "docker.io/ocpqe/hello-pod:latest"
+      imagePullPolicy: IfNotPresent
+      name: hello-pod
+      ports:
+        - containerPort: 8080
+          protocol: TCP
+      resources: {}
+      securityContext:
+        capabilities: {}
+        privileged: false
+      terminationMessagePath: /dev/termination-log
+  dnsPolicy: ClusterFirst
+  restartPolicy: Always
+  nodeSelector: <1>
+    env: test
+    os: fedora
+  serviceAccount: ""
+status: {}
+----
++
+<1> Labels as node selectors.
+
+
+. Create the pod in the project:
++
+----
+oc create -f pod.yaml --namespace=ns1
+----
+
+. Check that the node selector labels were added to the pod configuration:
++
+----
+get pod pod1 --namespace=ns1 -o json
+
+nodeSelector": {
+ "env": "test",
+ "infra": "fedora",
+ "os": "fedora"
+}
+----
++
+The node selectors are merged into the pod and the pod should be scheduled in the appropriate project.
+
+If you create a pod with a label that is not specified in the project specification, here `env: production`, the pod is not scheduled on the node:
++
+----
+nodeSelector:
+ "env: production"
+ "infra": "fedora",
+ "os": "fedora"
+----
+
+If there is a node that does not have a node selector annotation, the pod will be scheduled there. If there is no node
+
```

Comment:
seems not finished.
"If there is a node that does not have a node selector annotation, the pod will be scheduled there. If there is no node"